components/catalogs/front/src/hooks/useChannelsByCodes.ts

```diff
@@ -0,0 +1,27 @@
+import {useQuery} from 'react-query';
+import {Channel} from '../models/Channel';
+
+type ResultError = Error | null;
+type Result = {
+    isLoading: boolean;
+    isError: boolean;
+    data: Channel[] | undefined;
+    error: ResultError;
+};
+
+export const useChannelsByCodes = (codes: string[] | undefined): Result => {
```

Comment:
```suggestion
export const useChannelsByCodes = (codes: string[]): Result => {
```

IMO, you should request an array and not accept undefined. It should the calling component that should fallback on an empty array if its list of channels is undefined.